pkg/solana/chainreader/account_read_binding.go

```diff
@@ -22,14 +22,66 @@ type accountReadBinding struct {
 	account    solana.PublicKey
 	codec      types.RemoteCodec
 	reader     BinaryDataReader
+
+	chLoadedBts chan []byte
+	chLoadErr   chan error
+}
+
+func newAccountReadBinding(acct string, codec types.RemoteCodec, reader BinaryDataReader) *accountReadBinding {
+	return &accountReadBinding{
+		idlAccount:  acct,
+		codec:       codec,
+		reader:      reader,
+		chLoadedBts: make(chan []byte, 1),
+		chLoadErr:   make(chan error, 1),
+	}
 }
 
 var _ readBinding = &accountReadBinding{}
 
-func (b *accountReadBinding) GetLatestValue(ctx context.Context, _ any, outVal any) error {
+func (b *accountReadBinding) PreLoad(ctx context.Context) {
+	b.clearLoaded()
```

Comment:
Why do these need to be cleared each time? Are they re-used?

If this is necessary, then I think there is a race since this is called from a goroutine which could be delayed an arbitrary amount of time. So if there was something to clear, it is possible that the second for loop receives from the channel on the main goroutine before this clear call happens, delivering the data that was meant to be cleared.

Reply:
Instead of maintaining a single result/error channel in the binding, a result struct is provided to both `PreLoad` and `GetLatestValue` on the binding. This allows both calls to be linked by a single result struct that is isolated from other calls either in the same loop or in a different loop.